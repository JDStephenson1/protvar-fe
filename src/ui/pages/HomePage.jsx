
import React, { Fragment } from 'react';

import DefaultPageLayout from '../layout/DefaultPageLayout';
import SimpleSearch from '../components/search/SimpleSearch';



// class __HomePageContent extends Component {

//   render() {

//     const { searchTerm, searchResults } = this.state;
//     const rows = (null !== searchResults && searchResults.proteins)
//       ? searchResults.proteins
//       : {};
//     const ebiSearch = document.querySelector('#ebi-standard-search-field');
//     const ebiSearchField = document.querySelector('#ebi-standard-search-field #query');

//     if (ebiSearchField) {
//       ebiSearchField.value = searchTerm;
//     }

//     if (null !== searchResults) {
//       ebiSearch.style.display = 'block';
//     }

//     return(
//       <Fragment>
//         { null === searchResults
//           ? <SimpleSearch onSubmit={this.props.handleSearch} />
//           : <SearchResults rows={rows} />
//         }
//       </Fragment>
//     )
//   }
// };

const HomePageContent = props => {
  return (
    <Fragment>
      <SimpleSearch onSubmit={props.handleSearch} />
    </Fragment>
  );
};

const HomePage = props => (
  <DefaultPageLayout
    title="Home Page"
    content={<HomePageContent {...props} />}
  />
);

export default HomePage;
