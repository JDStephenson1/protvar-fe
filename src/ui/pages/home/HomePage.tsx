import DefaultPageLayout from '../../layout/DefaultPageLayout';
import FileUpload from './FileUpload';
import PasteVariantSearch from './PasteVariantSearch';
import { FileLoadFun } from '../../../utills/AppHelper';
import {Assembly, StringVoidFun} from '../../../constants/CommonTypes';
import { Link } from 'react-router-dom';
import { ABOUT, CONTACT } from '../../../constants/BrowserPaths';
import {API_URL} from "../../../constants/const";

const HomePageContent = (props: HomePageProps) => {
  const { loading, assembly, updateAssembly, fetchFileResult, fetchPasteResult } = props;

  return <>
    <div>
      <div className="banner">
        ProtVar will be launched on <strong>26th May</strong> at an EMBL-EBI webinar at 15:30 BST (UTC+1).
        We will discuss the various ways in which ProtVar can help users with their work as well as recent improvements and fixes.
        Places are limited so please register to secure your place <a href="https://www.ebi.ac.uk/training/events/contextualise-and-interpret-human-missense-variation-protvar/" target="_blank">here.</a>
      </div>

      <p>
        ProtVar (<strong>P</strong>rotein <strong>V</strong>ariation) is a resource to investigate missense variation in humans by presenting annotations which may be relevant to interpretation.
        <br/>
        Variants can be submitted below in genomic or protein formats, uploaded or accessed via our <Link to="" onClick={() => window.open(API_URL + '/docs', '_blank')} title="ProtVar REST API" target="_blank"  rel='noreferrer'> API </Link>.
      </p>
    </div>
    <div className="wrapper">
      <PasteVariantSearch isLoading={loading} assembly={assembly} updateAssembly={updateAssembly} fetchPasteResult={fetchPasteResult} />
      <FileUpload isLoading={loading} fetchFileResult={fetchFileResult} />
    </div>
    <div>
      <br />
      <p className="info">
        Further information can be found in the <Link to={ABOUT} title="About ProtVar's" className="ref-link">ABOUT</Link> section.
        Please <Link to={CONTACT} title="Contact us" className="ref-link">CONTACT</Link> us with specific queries or suggestions.
      </p>
    </div>
  </>
};


interface HomePageProps {
  loading: boolean
  assembly: Assembly
  updateAssembly: (assembly: Assembly) => void
  fetchFileResult: FileLoadFun
  fetchPasteResult: StringVoidFun
}
const HomePage = (props: HomePageProps) => <DefaultPageLayout content={<HomePageContent {...props} />} />;
export default HomePage;
