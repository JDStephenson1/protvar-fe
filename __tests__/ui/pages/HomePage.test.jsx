
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import HomePage from '../../../src/ui/pages/HomePage';

describe('HomePage component', () => {
  test('should render', () => {
    const props = {
      handleSearch: () => null,
      handleDownload: () => null,
    };

    const component = renderer
      .create(<MemoryRouter><HomePage {...props} /></MemoryRouter>)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
