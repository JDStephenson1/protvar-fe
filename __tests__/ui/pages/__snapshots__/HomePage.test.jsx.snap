// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component should render 1`] = `
Array [
  <div
    id="skip-to"
  >
    <a
      href="#content"
    >
      Skip to main content
    </a>
  </div>,
  <header
    className="clearfix masthead-black-bar"
    id="masthead-black-bar"
  >
    <nav
      className="row"
    >
      <ul
        className="menu global-nav text-right"
        id="global-nav"
      >
        <li
          className="home-mobile"
        >
          <a
            href="//www.ebi.ac.uk"
          >
            EMBL-EBI
          </a>
        </li>
        <li
          className="home"
        >
          <a
            href="//www.ebi.ac.uk"
          >
            EMBL-EBI
          </a>
        </li>
        <li
          className="services"
        >
          <a
            href="//www.ebi.ac.uk/services"
          >
            Services
          </a>
        </li>
        <li
          className="research"
        >
          <a
            href="//www.ebi.ac.uk/research"
          >
            Research
          </a>
        </li>
        <li
          className="training"
        >
          <a
            href="//www.ebi.ac.uk/training"
          >
            Training
          </a>
        </li>
        <li
          className="about"
        >
          <a
            href="//www.ebi.ac.uk/about"
          >
            About us
          </a>
        </li>
        <li
          className="float-right show-for-medium embl-selector embl-ebi active"
          id="embl-selector"
        >
          <button
            className="button float-right"
          >
            Â 
          </button>
        </li>
      </ul>
    </nav>
  </header>,
  <div
    className="content"
    id="content"
  >
    <div
      data-sticky-container={true}
    >
      <div
        className="masthead"
        data-btm-anchor="main-content-area:bottom"
        data-sticky={true}
        data-sticky-on="large"
        data-top-anchor="main-content-area:top"
        id="masthead"
      >
        <div
          className="masthead-inner row"
        >
          <div
            className="navbar"
          >
            <table>
              <tbody>
                <tr
                  className="navbar"
                >
                  <td>
                    <a
                      className="local-title"
                      href="/uniprot/pepvep/test/"
                      title="Back to PepVEP's homepage"
                    >
                      <i
                        className="fa fa-fw fa-home"
                      />
                       PepVEP
                    </a>
                    <a
                      className="local-sub-title"
                      href="/uniprot/pepvep/test/"
                      title="Back to PepVEP's homepage"
                    >
                      Predicting the effect of varying amino acids in proteins
                    </a>
                  </td>
                  <td
                    className="topnav-right local-sub-title"
                  >
                    <div>
                      <button
                        className="button"
                        onClick={[Function]}
                        type="button"
                      >
                        About
                      </button>
                    </div>
                  </td>
                  <td
                    className="topnav-right local-sub-title"
                  >
                    <div>
                      <button
                        className="button"
                        onClick={[Function]}
                        type="button"
                      >
                        Contact
                      </button>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
    <section
      className="row"
      role="main"
    >
      <div
        className="main-content-area row"
        id="main-content-area"
      >
        <div
          className="small-12 columns"
        >
          <div
            className="default-page-layout"
          >
            <div
              className="default-page-conent"
            >
              <div>
                <p>
                  PepVEP is an online service to interpret the effects of missense variants using protein function and structure. It utilises functional information from the Ensembl Variant Effect Predictor (VEP), the UniProt functional residue annotation (Protein function), and the PDBe structural residue annotation.
                  <br />
                </p>
                <p
                  className="info"
                >
                  Variants can be submitted via pasting in the box in VCF or HGVS format, uploading a file in VCF format or using the PepVEP API
                </p>
              </div>
              <div
                className="wrapper"
              >
                <div
                  className="card-table search"
                  id="search"
                >
                  <div
                    className="card"
                  >
                    <section
                      className="card__actions"
                    >
                      <span
                        className="card-header"
                      >
                        <p>
                          <b>
                            Paste variants (GRCh38)
                          </b>
                        </p>
                      </span>
                    </section>
                    <section
                      className="card--has-hover top-row"
                      role="button"
                    >
                      <div
                        className="card__content"
                      >
                        <section
                          className="uniprot-card"
                        >
                          <section
                            className="uniprot-card__left"
                          >
                            <textarea
                              className="main-textarea-search-field"
                              id="main-textarea-search-field"
                              onChange={[Function]}
                              onFocus={[Function]}
                              placeholder="Paste variants in HGVS or VCF format"
                              value=""
                            />
                            <table
                              className="table-input"
                            >
                              <tbody>
                                <tr>
                                  <td>
                                    <b>
                                      Examples:
                                    </b>
                                  </td>
                                  <td>
                                    <a
                                      onClick={[Function]}
                                    >
                                      VCF
                                    </a>
                                  </td>
                                  <td>
                                    <a
                                      onClick={[Function]}
                                    >
                                      HGVS
                                    </a>
                                  </td>
                                </tr>
                              </tbody>
                            </table>
                            <span
                              className="genome-assembly-text"
                            >
                              <p>
                                Reference Genome Assembly GRCh38 (hg38): 
                                     
                                <a
                                  className="ref-link"
                                  href="http://www.ensembl.org/Homo_sapiens/Tools/AssemblyConverter?db=core"
                                  rel="noopener noreferrer"
                                  target="_blank"
                                >
                                  Ensembl's Assembly Remapping
                                </a>
                              </p>
                            </span>
                            <div
                              className="search-button-group"
                              id="search-button-group"
                            >
                              <button
                                className="button-new button-primary button-bottom"
                                data-icon="="
                                onClick={[Function]}
                                type="submit"
                              >
                                Run
                              </button>
                            </div>
                          </section>
                        </section>
                      </div>
                    </section>
                  </div>
                </div>
                <div
                  className="card-table upload"
                  id="upload"
                >
                  <div
                    className="card"
                  >
                    <section
                      className="card__actions"
                    >
                      <span
                        className="card-header"
                      >
                        <p>
                          <b>
                            File Upload
                          </b>
                        </p>
                      </span>
                    </section>
                    <section
                      className="card--has-hover"
                      role="button"
                    >
                      <div
                        className="card__content"
                      >
                        <section
                          className="uniprot-card"
                        >
                          <section
                            className="uniprot-card__left"
                          >
                            <p>
                              <b>
                                PepVEP will interpret only the first five fields of the VCF
                              </b>
                              <br />
                              #CHROM POS ID REF ALT
                              <br />
                              Missing values should be specified with a dot (â.â)
                               
                              <a
                                href="https://www.ebi.ac.uk/training/online/courses/human-genetic-variation-introduction/variant-identification-and-analysis/understanding-vcf-format/"
                                target="_blank"
                              >
                                more info
                              </a>
                              <br />
                              <br />
                              <b>
                                PepVEP also supports hgvs in below format
                              </b>
                              <br />
                              NC_000010.11:g.121479868C&gt;G
                              <br />
                            </p>
                            <div
                              className="search-button-group"
                              id="search-button-group"
                            >
                              <input
                                id="myInput"
                                onChange={[Function]}
                                style={
                                  Object {
                                    "display": "none",
                                  }
                                }
                                type="file"
                              />
                              <button
                                className="button-new button-primary button-bottom"
                                data-icon="="
                                onClick={[Function]}
                                type="button"
                              >
                                Upload File
                              </button>
                            </div>
                          </section>
                        </section>
                      </div>
                    </section>
                  </div>
                </div>
                <div
                  className="card-table api"
                  id="api"
                >
                  <div
                    className="card"
                  >
                    <section
                      className="card__actions"
                    >
                      <span
                        className="card-header"
                      >
                        <p>
                          <b>
                            PepVEP REST API
                          </b>
                        </p>
                      </span>
                    </section>
                    <section
                      className="card--has-hover"
                      role="button"
                    >
                      <div
                        className="card__content"
                      >
                        <section
                          className="uniprot-card"
                        >
                          <section
                            className="uniprot-card__left"
                          >
                            <b>
                              PepVEP's REST API is a programmatic way to obtain information from PepVEP via Simple URL-based queries
                            </b>
                            <br />
                            You can query:
                            <ul>
                              <li>
                                Whole genes/proteins/structures specific residue ranges in proteins or protein structures or
                                 
                              </li>
                              <li>
                                Genomic ranges in genes a list of variants using several different formats
                              </li>
                            </ul>
                            Click below for all PepVEP REST API documentation
                            <div
                              className="search-button-group"
                              id="search-button-group"
                            >
                              <button
                                className="button-new button-bottom"
                                data-icon="="
                                onClick={[Function]}
                                type="button"
                              >
                                PepVEP REST API
                              </button>
                            </div>
                          </section>
                        </section>
                      </div>
                    </section>
                  </div>
                </div>
              </div>
              <div>
                <br />
                <p
                  className="info"
                >
                  Further help and explanations about the data in PepVEP can be found in the "about" section at the top right hand side of the page.
                  <br />
                  <br />
                  We continually strive to make PepVEP clear and useful to our users, to contact PepVEP with questions or suggestions please use the "contact" link at the top of the page.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  <footer
    id="footer-target"
  >
    <div
      className="global-footer"
      id="global-footer"
    >
      <nav
        className="global-nav-expanded row"
        id="global-nav-expanded"
      />
      <section
        className="ebi-footer-meta row"
        id="ebi-footer-meta"
      />
    </div>
  </footer>,
]
`;
